### PR TITLE
ros-canopen: adding can-msgs, socketcan-interface and socketcan-bridge

### PR DESCRIPTION
**Pull request by DasRoteSkelett**
Hi,

I added recipes for socketcan ros packages. Hope, it is acceptable for merge...

Regards,
   
 Matthias

### CODE REVIEW

File: recipes-ros/ros-canopen/socketcan-bridge_0.7.6.bb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+DESCRIPTION = "Provides nodes to convert messages from SocketCAN to a ROS Topic and vice versa."
+SECTION = "devel"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://package.xml;beginline=10;endline=10;md5=d566ef916e9dedc494f5f793a6690ba5"
+
+DEPENDS = "socketcan-interface can-msgs roscpp"
+
+
```
Comment (bulwahn) [inline recipes-ros/ros-canopen/socketcan-bridge_0.7.6.bb]: One empty line is sufficient

File: recipes-ros/ros-canopen/socketcan-bridge_0.7.6.bb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+require ros-canopen.inc
+
+ROS_SP="socketcan_bridge"
```
Comment (bulwahn) [inline recipes-ros/ros-canopen/socketcan-bridge_0.7.6.bb]: There is no newline at the EOF.

File: recipes-ros/ros-canopen/socketcan-interface_0.7.6.bb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+DESCRIPTION = "This package contains a generic CAN interface description with helpers for filtering and driver implementation. Further a socketcan implementation based on boost::asio is included."
+SECTION = "devel"
+LICENSE = "LGPLv3"
+LIC_FILES_CHKSUM = "file://package.xml;beginline=10;endline=10;md5=b691248d2f70cdaeeaf13696ada5d47c"
+
+DEPENDS = "boost class-loader virtual/kernel"
+
+
```
Comment (bulwahn) [inline recipes-ros/ros-canopen/socketcan-interface_0.7.6.bb]: One empty line is sufficient.

File: recipes-ros/ros-canopen/socketcan-interface_0.7.6.bb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+require ros-canopen.inc
+
+ROS_SP="socketcan_interface"
```
Comment (bulwahn) [inline recipes-ros/ros-canopen/socketcan-interface_0.7.6.bb]: There is no newline at the EOF.

File: recipes-ros/ros-canopen/can-msgs_0.7.6.bb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://package.xml;beginline=10;endline=10;md5=d566ef916e9dedc494f5f793a6690ba5"
+
+DEPENDS = "std-msgs genmsg message-generation"
+
+RDEPENDS_${PN} = "std-msgs message-runtime"
+
+require ros-canopen.inc
+
+ROS_SP="can_msgs"
```
Comment (bulwahn) [inline recipes-ros/ros-canopen/can-msgs_0.7.6.bb]: I am wondering why you set ROS_SP here? That does not follow the general pattern.

**Comment by bulwahn** [comment]
Thanks for your contribution. I will also run a quick test build and once the issues are resolved, we can merge it.

**Comment by DasRoteSkelett** [comment]
Hi,

I added another commit, you were right, ROS_SP unneeded!

Regards,
  Matthias

**Comment by DasRoteSkelett** [comment]
Argh, the last commit has nothing to do with the socketcan, it is bumping versions to kinetic ... just in case you are wondering...

**Comment by bulwahn** [comment]
The last patch totally makes sense, I just did not come across that because the experimental branch is just continuous work in progress and it has not yet stabilized and goes through a lot of testing. I think after I reviewed patch 1 & 2, you can actually now squash them (with `git rebase -i`).
The third patch would be better if you could split that in three patches:
  - One patch where you just replace all incidences of the hard-coded string indigo in the recipes/classes by `${ROSDISTRO}`. This, we can then also include in the indigo branch as well as general cleanup.
  - and a second patch where you change the `${ROSDISTRO}` to `kinetic`.
  - and a third patch where you update the bfl recipe.

So, what do you think? For the second patch, you can open a pull request on the indigo branch at the bmwcarit repository as well.

**Comment by DasRoteSkelett** [comment]
I will do as recommended

**Pull request closed without merging**